### ISSUE TITLE
Please support authinfo for login

### ISSUE THREAD

**Issue by xmaillard** [nlamirault/emacs-gitlab#28]
Hello,

it would be awesome to protect our login information inside a GPG protected file like the .authinfo.gpg file.

Thank you.
Xavier

**Comment by xmaillard** [comment]
Here is a sample way it could be done (https://git.framasoft.org/snippets/120):

```lisp
(defun gitlab--get-password-authinfo (&rest username)
  "Return password for `username' using `authinfo' file."
  (let* ((username (or username (gitlab--get-username)))
	 (host (gitlab--get-host))
	 (found (car (auth-source-search
			:host host :require '(:secret) :user username))))
    (when found
      (let ((password (plist-get found :secret)))
	(if (functionp password) ;; could be a function check `auth-source-search'
	    (funcall password)
	  password)))))

(setq gitlab-host "git.framasoft.org"
      gitlab-username "xma"
      gitlab-password (gitlab--get-password-authinfo))
```

Note: no password cache mechanism

Authinfo could be used also to store the token in return, etc. I feel it more secure than any other mechanism (authinfo can be encrypted).

**Comment by thomasf** [comment]
you could also just have a `gitlab-auth.el.gpg` and load that when you require emacs-gitlab.. You don.'t need something special case functionality for that.

**Comment by xmaillard** [comment]
True. Though I prefer to have all my auth stuff in one (protected) place. YMMV :)

There is also the password-store CLI (which I have not checked yet).

**Comment by TLATER** [comment]
Authinfo is the standard for everything else in emacs, why not support that? Authinfo also allows using different backends, making it a lot more convenient for users than encrypting .el files.
